src/renderer/features/governance/model/locks.ts

```diff
@@ -1,77 +1,68 @@
+const $isLoading = createStore(true).reset(networkSelectorModel.events.chainChanged || walletModel.$activeWallet);
 
-const $maxLock = createStore<BN>(BN_ZERO);
-const $chainId = restore(chainIdChanged, null);
-const $isLocksRequested = createStore(false).reset(chainIdChanged);
-const $isLoading = createStore(true).reset(chainIdChanged);
+type Props = {
+  api: ApiPromise;
+  addresses: Address[];
+};
 
-const getClassLocksFx = createEffect((api: ApiPromise): Promise<Record<TrackId, BN>> => {
-  // return governanceService.getTrackLocks(api, '12mP4sjCfKbDyMRAEyLpkeHeoYtS5USY4x34n9NMwQrcEyoh');
-  return governanceService.getTrackLocks(api, '153YD8ZHD9dRh82U419bSCB5SzWhbdAFzjj4NtA5pMazR2yC');
+const getTrackLocksFx = createEffect(({ api, addresses }: Props): Promise<Record<Address, Record<TrackId, BN>>> => {
+  return governanceService.getTrackLocks(api, addresses);
 });
 
-const $asset = combine(
-  {
-    chainId: $chainId,
-    chains: networkModel.$chains,
-  },
-  ({ chainId, chains }) => {
-    return (chainId && chains[chainId].assets[0]) || null;
-  },
-);
-
-const $api = combine(
-  {
-    chainId: $chainId,
-    apis: networkModel.$apis,
-  },
-  ({ chainId, apis }) => {
-    return (chainId && apis[chainId]) || null;
-  },
-);
+const $asset = networkSelectorModel.$governanceChain?.map((chain) => (chain && chain.assets[0]) || null);
 
 sample({
-  clock: $api.updates,
-  source: $isLocksRequested,
-  filter: (isLocksRequested, api) => !isLocksRequested && Boolean(api),
-  fn: (_, api) => api!,
-  target: getClassLocksFx,
+  clock: [networkSelectorModel.$governanceChainApi, walletModel.$activeWallet],
+  source: {
+    chain: networkSelectorModel.$governanceChain,
+    api: networkSelectorModel.$governanceChainApi,
+    wallet: walletModel.$activeWallet,
+  },
+  filter: ({ chain, api, wallet }) => !!chain && !!api && !!wallet,
+  fn: ({ api, chain, wallet }) => ({
+    api: api!,
+    addresses: accountUtils.getAddressesForWallet(wallet!, chain!),
+  }),
+  target: getTrackLocksFx,
 });
 
 sample({
-  clock: getClassLocksFx.doneData,
+  clock: getTrackLocksFx.doneData,
   fn: (trackLocks) => {
-    const locks = Object.values(trackLocks).reduce<BN>((acc, lock) => (lock.gt(acc) ? lock : acc), BN_ZERO);
+    let maxLockTotal = BN_ZERO;
+    for (const address in trackLocks) {
+      const lock = trackLocks[address];
+      const totalLock = Object.values(lock).reduce<BN>((acc, lock) => (lock.gt(acc) ? lock : acc), BN_ZERO);
+
+      maxLockTotal = maxLockTotal.add(totalLock);
+    }
 
-    return { locks, requested: true };
+    return { maxLockTotal, trackLocks };
   },
-  target: spread({
-    locks: $maxLock,
-    requested: $isLocksRequested,
-  }),
+  target: spread({ maxLockTotal: $totalLock, trackLocks: $trackLocks }),
 });
 
 sample({
-  clock: getClassLocksFx.finally,
+  clock: getTrackLocksFx.finally,
   fn: () => false,
   target: $isLoading,
 });
 
 export const locksModel = {
-  $maxLock,
   $asset,
   $isLoading,
```

Comment:
Looks like `$isLoading: getTrackLocksFx.pending` enough here? But I am not sure for 100%